src/main/scala/reactor/core/scala/publisher/SFlux.scala

```diff
@@ -1149,7 +1149,12 @@ object SFlux {
 
   def push[T](emitter: FluxSink[T] => Unit, backPressure: FluxSink.OverflowStrategy = OverflowStrategy.BUFFER): SFlux[T] = new ReactiveSFlux[T](JFlux.push(emitter, backPressure))
 
-  def raiseError[T](e: Throwable, whenRequested: Boolean = false): SFlux[T] = new ReactiveSFlux[T](JFlux.error(e, whenRequested))
+  @deprecated("Use error(Throwable, Boolean) instead")
+  def raiseError[T](exception: Throwable, whenRequested: Boolean = false): SFlux[T] = error(exception)
+
+  def error[T](exception: Throwable): SFlux[T] = new ReactiveSFlux[T](JFlux.error(exception))
```

Comment:
Do we need this variant? Can still use false as the default value like raiseError did.

On the other hand, if revisiting these method signatures, it might be a good time to consider adding a lazy by-name variant that calls the JFlux/JMono Supplier<Throwable> overload

Reply:
To be honest I wouldn't provide these deferred overloads of the error method at all.
I don't see any use case to change the error type depending on the subscription time.
Even the just method doesn't have this deferred overloads. 

You cannot have the by name and the by value variants at the same time. You would need to make it a none arg function like so: 
`def error(e: ()=> Throwable)` 

But still if there is one of the rare use cases where the emitted error depends on the subscription  time one could use create() or generate(). 
I would be in favour of having just a the error(e: Throwable) variant.

Reply:
To be honest, I can't think of any scenario that would differentiate between raising the error when being subscribed or when being requested (that's the meaning of `whenRequested` parameter). However, in spirit of the "principle of least surprise" as you put in the beginning, I think we should retain the `whenRequested` parameter as that would pass the value down to `JFlux` if it is really needed.

Reply:
No problem, whatsoever. That's why I kept both overloads exactly as if directly inherited from Java. 
Unfortunately there was an compilation error in 2.11 variant which was not related to my changes. 
What can I do now in order the change in?

Reply:
> Unfortunately there was an compilation error in 2.11 variant which was not related to my changes.

@jpork I've removed support for 2.11 in master. If you get latest master, there is no more 2.11

Reply:
F*. If just reloved the compilation and test errors. I try to roll it back quickly.

Reply:
Ok, rolled back 2.11 fixes. I use the gradle buildAll task for build. The gradle build seems to still contain scala-2.11 build target.

Reply:
My build fails for this SFluxTest:
```
    ".publish" - {
      "without transformer should produce connectable flux" in {
        val buffer = mutable.ListBuffer.empty[Int]
        val base = SFlux.just(1, 2, 3, 4, 5).delayElements(1 second).publish().autoConnect()
        base.subscribe()
        Thread.sleep(1000)
        base.subscribe(i => buffer += i)
        eventually {
          buffer.size should be > 1
          buffer should not contain 1
        }
      }
    }
```
` Thread.sleep(1000)` could be a bit close to the limits when you shift elements by on seconds. Increasing it e.g. to 1200 let it succeed safely on my machine too.

Reply:
> Ok, rolled back 2.11 fixes. I use the gradle buildAll task for build. The gradle build seems to still contain scala-2.11 build target.

Ah, thanks... I've removed it already in master now

Reply:
> My build fails for this SFluxTest:
> 
> ```
>     ".publish" - {
>       "without transformer should produce connectable flux" in {
>         val buffer = mutable.ListBuffer.empty[Int]
>         val base = SFlux.just(1, 2, 3, 4, 5).delayElements(1 second).publish().autoConnect()
>         base.subscribe()
>         Thread.sleep(1000)
>         base.subscribe(i => buffer += i)
>         eventually {
>           buffer.size should be > 1
>           buffer should not contain 1
>         }
>       }
>     }
> ```
> 
> ` Thread.sleep(1000)` could be a bit close to the limits when you shift elements by one seconds. Increasing it e.g. to 1200 let it succeed safely on my machine too.

Thanks... I found this irritating as well but couldn't replicate in my own workstation.